qemu/qemu.go

```diff
@@ -1221,9 +1225,18 @@ func (config *Config) appendCPUs() {
 			SMPParams = append(SMPParams, fmt.Sprintf(",sockets=%d", config.SMP.Sockets))
 		}
 
+		if config.SMP.MaxCPUs > 0 {
```

Comment:
Is this test correct?  The help above states that MaxCPUs must be equal or greater than CPUs.  If we were to have CPUs = 1 and maxCPUS = 0, we would not report an error, even though we've violated the constraint.

Reply:
yes it is, I wouldn't like to break the code that already uses this function but don't include ```MaxCPUs``` in the ```SMP``` struct

Reply:
ohh wait, this is not ciao, forget it, I'll remove this condition :)

Reply:
ciao doesn't use this code anyway.

Reply:
But I take your point.  If you remove the check you'll need to be careful to update all the places in CC where the SMP struct is used and CPUs is specified.

Reply:
yep, I'll do

Reply:
@devimc You're going to hate me but maybe you were right to have the if statement.  Without it we do break backwards compatibility.   Maybe not a problem right now, as there are only 3 users of this library, but it doesn't really set a great precedent.  How about we add the if statement back in and update the help for MaxCPUs.

from

This value MUST BE equal to or greater than CPUs

to

This value, if non-zero, MUST BE equal to or greater than CPUs

That was really point of my original comment, that the code didn't match the help.

Reply:
ha, ok 👍